source/AI.cpp

```diff
@@ -141,6 +143,36 @@ namespace {
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 
+	// Helper function for selecting the ships for formation commands.
+	vector<Ship *> GetShipsForFormationCommand(const PlayerInfo &player)
+	{
+		// Figure out what ships we are giving orders to.
+		vector<Ship *> targetShips;
+		auto &selectedShips = player.SelectedShips();
+		bool fullFleet = selectedShips.empty();
+		if(fullFleet)
+		{
+			auto &playerShips = player.Ships();
+			targetShips.reserve(playerShips.size() - 1);
+			for(const shared_ptr<Ship> &it : player.Ships())
+				if(it.get() != player.Flagship() && !it->IsParked()
+						&& !it->IsDestroyed())
```

Comment:
Don't we want to exclude disabled ships here?
```suggestion
						&& !it->IsDestroyed() && !it->IsDisabled())
```